adds documentation for getting start with the scaffolder (#494)
RoadieHQ/marketing-site: content/docs/getting-started/scaffolding-components/index.md

---
title: Scaffolding Components in Backstage
lastUpdated: '2021-11-24T14:00:00.0Z'
description: How to create components such as services or websites using the Backstage scaffolder.
---

## Introduction

This tutorial will show you how to create a basic scaffolder template in Backstage.

> ## Prerequisite 

Reviewer: You have to enable it in the settings too right? An admin could turn off the scaffolder for a tenant, then another user of the same tenant could read this and be confused.

Author: The first step explains how to enable it. Are you suggesting that I move that up to the prerequisites?

Just not sure what you mean.

Reviewer: Ah yes. Indeed it does!

You must have installed the Github App in order to use the scaffolder. If you have not done this you can find the steps to do this [here.](https://roadie.io/docs/getting-started/getting-started-for-admins/#connect-roadie-to-github) 

## Steps

You can either follow the text or the video, or both.

<div style="position: relative; padding-bottom: 52.42718446601942%; height: 0;"><iframe src="https://www.loom.com/embed/da6159c4ca39458cb5ad03138612a5a3" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>

### Step One. Enable the scaffolder

Visit the `Administration > Settings > Scaffolder` and make sure the enabled check box is selected, press save and apply and restart.

### Step Two. Create a template.yaml file in a repository

> You can either create the following file in your own repository or use our example that is in a public repository. https://github.com/RoadieHQ/sample-service/blob/main/demo_template.yaml

Reviewer: ```suggestion
You can either create the following file in your own repository or [use the public example](https://github.com/RoadieHQ/sample-service/blob/main/demo_template.yaml) we prepared. 
```

<details>

>   <summary>Expand here to see an example template.yaml</summary>

Reviewer: Love that this is hidden until required. Very nice touch!

Author: Thank you
